src/prefect/cli/heartbeat.py

```diff
@@ -89,10 +91,32 @@ def flow_run(id, num):
     """
 
     client = Client()
+    logger = get_logger("heartbeat")
     iter_count = 0
 
-    while iter_count < (num or 1):
-        client.update_flow_run_heartbeat(id)  # type: ignore
-        if num:
-            iter_count += 1
-        time.sleep(config.cloud.heartbeat_interval)
+    # Ensure that logs are sent to the backend since this is typically called without
+    # console stdout/err
+    with prefect.context({"flow_run_id": id, "running_with_backend": True}):
+
+        try:  # Log signal-like exceptions that cannot be ignored
+
+            while iter_count < (num or 1):
+
+                try:  # Ignore (but log) client exceptions
+                    client.update_flow_run_heartbeat(id)
+                except Exception as exc:
+                    logger.error(
+                        f"Failed to send heartbeat with exception: {exc!r}",
```

Comment:
Does repr'ing the exception cause the log message to be twice as long?

Reply:
Hm; it depends on the exception but most of what I'm looking for with `exc_info` is the traceback. It looks like

```
└── 15:32:43 | ERROR   | Failed to send heartbeat with exception: ValueError('Foo')
Traceback (most recent call last):
  File "/Users/mz/prefect/core/src/prefect/cli/heartbeat.py", line 107, in flow_run
    raise ValueError("Foo")
ValueError: Foo
```

Might be a little overkill but 🤷 I'll keep that in mind for future error logs.